lib/internal/queue.js

```diff
@@ -46,7 +47,11 @@ export default function queue(worker, concurrency, payload) {
                 q._tasks.push(item);
             }
         }
-        setImmediate(q.process);
+
+        if (!isWaitingForProcessing) {
+            isWaitingForProcessing = true;
+            setImmediate(q.process);
+        }
```

Comment:
I wouldn't worry about the `q.process` change.  It's a footgun if a user messes with it.